sdk/examples/wallet/multithread.rs

```diff
@@ -0,0 +1,198 @@
+        .finish()
+        .await?;
+
+    let alias1 = var("ACCOUNT_ALIAS_1").unwrap();
+    let account1 = get_or_create_account(&wallet, &alias1).await?;
+    let alias2 = var("ACCOUNT_ALIAS_2").unwrap();
+    let account2 = get_or_create_account(&wallet, &alias2).await?;
+
+    let account1_send_address = &account1.addresses().await?[0];
+    let account2_recv_addresses = generate_addresses(&account2).await?;
+
+    sync_and_print_balance(&account1).await?;
+    sync_and_print_balance(&account2).await?;
+
+    may_request_funds(&account1, &account1_send_address.address().to_string()).await?;
+
+    let mut tasks: JoinSet<Result<(usize, usize)>> = JoinSet::new();
+    let num_threads_per_address = num_cpus::get().min(MAX_CPUS_TO_USE);
+
+    for address_index in 0..NUM_RECV_ADDRESSES {
+        for thread_index in 1..=num_threads_per_address {
+            let account1_clone = account1.clone();
+            let account2_recv_addresses_clone = account2_recv_addresses.clone();
+
+            tasks.spawn(async move {
+                let amount = ((address_index + thread_index) % 3 + 1) as u64 * BASE_AMOUNT;
+                let recv_address = account2_recv_addresses_clone[address_index % NUM_RECV_ADDRESSES].address();
+                println!("Sending '{amount}' coins to '{recv_address}'...");
+
+                let transaction = if (address_index + thread_index) % 2 == 0 {
+                    // ALTERNATIVE 1: using `account.send_amount``
+                    let outputs = vec![iota_sdk::wallet::SendAmountParams::new(
+                        recv_address.to_string(),
+                        amount,
+                    )];
+                    account1_clone.send_amount(outputs, None).await?
+                } else {
+                    // ALTERNATIVE 2: using `account.send`
+                    let outputs = vec![
+                        iota_sdk::types::block::output::BasicOutputBuilder::new_with_amount(amount)
+                            .add_unlock_condition(
+                                iota_sdk::types::block::output::unlock_condition::AddressUnlockCondition::new(
+                                    *recv_address.as_ref(),
+                                ),
+                            )
+                            .finish_output(account1_clone.client().get_token_supply().await?)?,
+                    ];
+                    account1_clone.send(outputs, None).await?
+                };
+
+                println!(
+                    "Transaction to address {} from thread {thread_index}/{num_threads_per_address} sent: {}",
+                    recv_address, transaction.transaction_id
+                );
+
+                // Wait for transaction to get included
+                let block_id = account1_clone
+                    .retry_transaction_until_included(&transaction.transaction_id, None, None)
+                    .await?;
+
+                println!(
+                    "Transaction included: {}/block/{}",
```

Comment:
lets add thread index/num here as well

Reply:
this example actually didn't work as intended, so I deleted it. It really doesn't add much anyway when compared with the `spammer` example, and there ... the multithreading actually works.